frontend/src/containers/TaskForm/index.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { loadAllTasksAction, loadAllTasksFailAction, loadAllTasksSuccessAction, addNewTaskAction } from '../../stores/actions';
+import { selectAllTasks, selectAuthToken, selectIsAllTasksLoaded } from '../../stores/selectors';
+import { updateNewTask } from '../../remotes/api';
+import Dropdown from '../../components/Dropdown'
+
+async function postNewTask(authToken, newTask) {
+  if (!newTask) {
+    return null;
+  }
+
+  const dispatch = useDispatch();
+
+  updateNewTask(authToken, newTask)
+    .then(() => {
+      dispatch(addNewTaskAction(authToken, newTask));
+    })
+    .catch(error => {
+      dispatch(loadAllTasksFailAction(error));
+    });
+}
```

Comment:
백엔드 서버로 newTask를 업데이트하는 api를 연동하고 동시에 리덕스 스토어에도 값을 변경하고자 했는데,
이렇게 하나의 함수로 정의하여 사용해도 괜찮을까요?
(참고로 postNewTask는 `< Dropdown />` 컴포넌트에 prop으로 전달됩니다)

Reply:
음 이렇게 하면 `useDispatch`가 작동안할 겁니다ㅠㅠ 훅은 *반드시* 함수형 컴포넌트 내에서만 사용 가능해요.

```suggestion
const postNewTask = useCallback(async (newTask) => {
  if (!newTask) {
    return;
  }
  
  updateNewTask(authToken, newTask)
      .then(() => {
        dispatch(addNewTaskAction(authToken, newTask));
      })
      .catch(error => {
        dispatch(loadAllTasksFailAction(error));
      });
}, [dispatch, authToken]);
```

요런식으로 콜백 훅을 컴포넌트에 넣고 써보시겠어요?

Reply:
넵, 감사합니다! 이제 dispatch가 제대로 작동하네요 👍